app/grandchallenge/reader_studies/views.py

```diff
@@ -867,6 +932,107 @@ def ground_truth(self, request, pk=None, case_pk=None):
         )
 
 
+class DisplaySetViewSet(
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.ListModelMixin,
+    GenericViewSet,
+):
+    serializer_class = DisplaySetSerializer
+    permission_classes = [DjangoObjectPermissions]
+    filter_backends = [DjangoFilterBackend, ObjectPermissionsFilter]
+    filterset_fields = ["reader_study"]
+    renderer_classes = (
+        *api_settings.DEFAULT_RENDERER_CLASSES,
+        PaginatedCSVRenderer,
+    )
+
+    @property
+    def reader_study(self):
+        reader_study_pk = self.request.query_params.get("reader_study")
+        if reader_study_pk:
+            return ReaderStudy.objects.get(pk=reader_study_pk)
+
+    def get_serializer_class(self):
+        if self.action in ["partial_update", "update", "create"]:
+            return DisplaySetPostSerializer
+        return DisplaySetSerializer
+
+    def partial_update(self, request, pk=None):
+        instance = self.get_object()
+        if request.data.get("value"):
+            # Get the provided civ from the current reader study's display sets.
+            civs = instance.reader_study.display_sets.values_list(
+                "values", flat=True
+            )
+            civ = ComponentInterfaceValue.objects.filter(id__in=civs).get(
+                id=request.data.get("value")
+            )
+
+            # Because the civ is added to a display set, it can no longer be
+            # a value for another display set.
+            civ.display_sets.clear()
```

Comment:
Why not, what about it belonging to a display set of another reader study?

Reply:
Good point. It shouldn't be a part of any other display set in this reader study though right?

Reply:
I think that users want to re-use images within the same study, see the usage of "disable hanging list validation".

Reply:
Is that something we want to allow by default or should there be a flag for it like there is now?